core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -1593,6 +1594,22 @@ public void addView(View v) throws IOException {
         viewGroupMixIn.addView(v);
     }
 
+    /**
+     * Completely replaces views
+     */
+    @DataBoundSetter
+    public void setViews(Collection<View> views) throws IOException {
```

Comment:
You are right that this is for system config DSL. `@DataBoundSetter` was unintended.

There's no meaningful synchronization that can be done here on `CopyOnWriteArrayList`, and I don't think it's a problem.